pipeline/special-case-executor.groovy

```diff
@@ -0,0 +1,186 @@
+                ]]
+            ])
+
+            // prepare the node
+            sharedLib = load("${env.WORKSPACE}/pipeline/vars/lib.groovy")
+            sharedLib.prepareNode()
+        }
+    }
+
+    stage('Prepare-Stages') {
+        /* Prepare pipeline stages using RHCEPH version */
+        ciMap = sharedLib.getCIMessageMap()
+        previousTierLevel = ciMap.test.type
+
+        tierTypeStrings = previousTierLevel.split("-")
+        def tierLevelInt = tierTypeStrings[1].toInteger() + 1
+        tierLevel = tierTypeStrings[0] + "-" + tierLevelInt
+
+        // Once the pipeline matures, uncomment the below line
+        // cliArg = "--build ${tierLevel}"
+        def cliArgs = "--build tier-0"
+        if ( ciMap.artifact.build == "signed" ) {
+            cliArgs = "--build rc"
+        }
+
+        def rhcsVersion = sharedLib.getRHCSVersionFromArtifactsNvr()
+        majorVersion = rhcsVersion["major_version"]
+        minorVersion = rhcsVersion["minor_version"]
+
+        /*
+           Read the release yaml contents to get contents,
+           before other listener/Executor Jobs updates it.
+        */
+        releaseContent = sharedLib.readFromReleaseFile(
+            majorVersion, minorVersion, lockFlag=false
+        )
+
+        // Till the pipeline matures, using the build that has passed tier-0 suite.
+        testStages = sharedLib.fetchStages(cliArgs, tierLevel, testResults)
+        testStages = testStages.findAll { !it.key.contains("psi-only")}
+
+        if ( testStages.isEmpty() ) {
+            currentBuild.result = "ABORTED"
+            error "No test stages found.."
+        }
+
+        currentBuild.description = "${ciMap.artifact.nvr} - ${ciMap.artifact.version} - ${tierLevel}"
+    }
+
+    // Running the test suites in batches of 4
+    (testStages.keySet() as List).collate(4).each{
+        def stages = testStages.subMap(it)
+        parallel stages
+    }
+
+    stage('Publish Results') {
+        /* Publish results through E-mail and Google Chat */
+
+        if ( ! ("FAIL" in sharedLib.fetchStageStatus(testResults)) ) {
+            def latestContent = sharedLib.readFromReleaseFile(
+                majorVersion, minorVersion
+            )
+            if ( ! releaseContent.containsKey(previousTierLevel) ) {
+                sharedLib.unSetLock(majorVersion, minorVersion)
+                error "No data found for pre tier level: ${previousTierLevel}"
+            }
+
+            if ( latestContent.containsKey(tierLevel) ) {
+                latestContent[tierLevel] = releaseContent[previousTierLevel]
+            } else {
+                def updateContent = [
+                    "${tierLevel}": releaseContent[previousTierLevel]
+                ]
+                latestContent += updateContent
+            }
+
+            sharedLib.writeToReleaseFile(majorVersion, minorVersion, latestContent)
+            println "latest content is: ${latestContent}"
+        }
```

Comment:
This not being the primary execution space... let's avoid updating the recipe files

Reply:
Done.